test/lsp.py

```diff
@@ -1445,8 +1452,11 @@ def test_generic(self, solc: JsonRpcProcess) -> None:
 
         for sub_dir in map(lambda filepath: filepath.name, sub_dirs):
             tests = map(
-                lambda filename, sd=sub_dir: sd + "/" + filename[:-len(".sol")],
-                os.listdir(f"{self.project_root_dir}/{sub_dir}")
+                lambda file_object, sd=sub_dir: sd + "/" + file_object.name[:-len(".sol")],
```

Comment:
Your suggestion won't work because `sub_dir` at the time of the lambda execution might have a completely different value, so we use a capture here to make the right `sub_dir` value available. See also https://docs.python.org/3/faq/programming.html#why-do-lambdas-defined-in-a-loop-with-different-values-all-return-the-same-result

Reply:
Ah, ok. That's something that at least warrants a comment though. It's not obvious.